### PR TITLE
Reapply [BOLT] DataAggregator support for binaries with multiple text segments

### PR DESCRIPTION
**Pull request by paschalis-mpeis**
When a binary has multiple text segments, the Size is computed as the
difference of the last address of these segments from the BaseAddress.
The base addresses of all text segments must be the same.

Introduces flag 'perf-script-events' for testing, which allows passing perf events
without BOLT having to parse them by invoking 'perf script'. The flag is used to
pass a mock perf profile that has two memory mappings for a mock binary
that has two text segments. The mapping size is updated as
`parseMMapEvents` now processes all text segments.

### CODE REVIEW

**Comment by zmodem** [comment]
> @zmodem, this should fix the reported failure.

Verified that it passes the tests locally for me. Thanks!

**Comment by paschalis-mpeis** [comment]
Can anyone approve this patch before I re-apply it? Thanks.

**Review by maksfb — APPROVED** [review]

**Comment by paschalis-mpeis** [comment]
Adding that the reported ASAN buildbot [failure](https://github.com/llvm/llvm-project/pull/118023#issuecomment-2511738211) should be unrelated to this patch.

File: llvm/utils/gn/secondary/bolt/unittests/Core/BUILD.gn [hunk 1]
```diff
@@ -15,6 +15,7 @@ unittest("CoreTests") {
     "BinaryContext.cpp",
     "DynoStats.cpp",
     "MCPlusBuilder.cpp",
+    "MemoryMaps.cpp",
```
Comment (nico) [inline llvm/utils/gn/secondary/bolt/unittests/Core/BUILD.gn:18]: (No need to change .gn files; they're updated automatically by a bot when possible.)

Comment (paschalis-mpeis) [reply llvm/utils/gn/secondary/bolt/unittests/Core/BUILD.gn:18]: Good to know. Thanks!

**Pull request merged**